src/main/java/com/softserve/academy/spaced/repetition/controller/CourseController.java

```diff
@@ -56,102 +55,102 @@
         Page<CourseLinkDTO> courseLinkDTOS = courseService
                 .getPageWithCourses(pageNumber, sortBy, ascending).map((course) -> {
                     Link selfLink = linkTo(methodOn(CourseController.class)
-                            .getCourseById(course.getCategory().getId(), course.getId())).withSelfRel();
+                            .getCourseById(course.getId())).withSelfRel();
                     return DTOBuilder.buildDtoForEntity(course, CourseLinkDTO.class, selfLink);
                 });
         return new ResponseEntity<>(courseLinkDTOS, HttpStatus.OK);
     }
 
-    @GetMapping(value = "/api/courses/ordered")
+    @GetMapping(value = "/api/courses/orders")
     public ResponseEntity<List<CourseLinkDTO>> getAllCoursesOrderByRating() {
         List<Course> courseList = courseService.getAllOrderedCourses();
-        Link collectionLink = linkTo(methodOn(CourseController.class).getAllCoursesOrderByRating()).withRel("course");
+        Link collectionLink = linkTo(methodOn(CourseController.class).getAllCoursesOrderByRating()).withRel("courses");
         List<CourseLinkDTO> decks = DTOBuilder
                 .buildDtoListForCollection(courseList, CourseLinkDTO.class, collectionLink);
         return new ResponseEntity<>(decks, HttpStatus.OK);
     }
 
     @Auditable(action = AuditingAction.VIEW_TOP_COURSES)
-    @GetMapping("/api/course/top")
+    @GetMapping("/api/courses/top")
     public ResponseEntity<List<CourseLinkDTO>> getTopCourse() {
         List<Course> courseList = courseService.getTopCourse();
         List<CourseLinkDTO> courses = new ArrayList<>();
         for (Course course : courseList) {
             Link selfLink = linkTo(methodOn(CourseController.class)
-                    .getCourseById(course.getCategory().getId(), course.getId())).withSelfRel();
+                    .getCourseById(course.getId())).withSelfRel();
             courses.add(DTOBuilder.buildDtoForEntity(course, CourseLinkDTO.class, selfLink));
         }
         return new ResponseEntity<>(courses, HttpStatus.OK);
     }
 
-    @GetMapping(value = "/api/category/{category_id}/courses/{course_id}")
-    @PreAuthorize(value = "@accessToUrlService.hasAccessToCourse(#category_id, #course_id)")
-    public ResponseEntity<CourseLinkDTO> getCourseById(@PathVariable Long category_id, @PathVariable Long course_id) {
-        Course course = courseService.getCourseById(category_id, course_id);
-        Link selfLink = linkTo(methodOn(CourseController.class).getCourseById(category_id, course_id)).withSelfRel();
+    @GetMapping(value = "/api/courses/{course_id}")
+    public ResponseEntity<CourseLinkDTO> getCourseById(@PathVariable Long course_id) {
+        Course course = courseService.getCourseById(course_id);
+        Link selfLink = linkTo(methodOn(CourseController.class).getCourseById(course_id)).withSelfRel();
```

Comment:
Im not sure, but if we get course, we have link to this course (self) and also we should have link to decks of this course (decks).

Reply:
all is good, I think